app/auth/middleware_test.go

```diff
@@ -1,43 +1,138 @@
 
-	"github.com/France-ioi/AlgoreaBackend/app/config"
+	"github.com/France-ioi/AlgoreaBackend/app/database"
+	"github.com/France-ioi/AlgoreaBackend/app/logging"
+	"github.com/France-ioi/AlgoreaBackend/app/loggingtest"
 )
 
-type authResp struct {
-	UserID int64  `json:"userID"`
-	Error  string `json:"error"`
+func TestUserIDMiddleware(t *testing.T) {
+	tests := []struct {
+		name                     string
+		authHeader               string
+		expectedAccessToken      string
+		userIDReturnedByDB       int64
+		dbError                  error
+		expectedStatusCode       int
+		expectedServiceWasCalled bool
+		expectedBody             string
+		expectedLogs             string
+	}{
+		{
+			name:                     "valid access token",
+			authHeader:               "Bearer 1234567",
+			expectedAccessToken:      "1234567",
+			userIDReturnedByDB:       890123,
+			expectedStatusCode:       200,
+			expectedServiceWasCalled: true,
+			expectedBody:             "user_id:890123",
+		},
+		{
+			name:                     "missing access token",
+			expectedStatusCode:       401,
+			expectedServiceWasCalled: false,
+			expectedBody:             "No access token provided",
+		},
+		{
+			name:                     "database error",
+			authHeader:               "Bearer 123",
+			expectedAccessToken:      "123",
+			dbError:                  errors.New("some error"),
+			expectedStatusCode:       502,
+			expectedServiceWasCalled: false,
+			expectedBody:             "Can't validate the access token",
+			expectedLogs:             `level=error msg="Can't validate an access token: some error"`,
+		},
+		{
+			name:                     "expired token",
+			authHeader:               "Bearer abcdefgh",
+			expectedAccessToken:      "abcdefgh",
+			expectedStatusCode:       401,
+			expectedServiceWasCalled: false,
+			expectedBody:             "The access token has expired",
+		},
+		{
+			name:                     "spaces before the access token",
+			authHeader:               "Bearer   1234567",
+			expectedStatusCode:       401,
+			expectedServiceWasCalled: false,
+			expectedBody:             "No access token provided",
+		},
+		{
+			name:                     "spaces in access token",
+			authHeader:               "Bearer 123 456 7",
+			expectedStatusCode:       401,
+			expectedServiceWasCalled: false,
+			expectedBody:             "No access token provided",
+		},
+		{
+			name:                     "ignores other kinds of authorization headers",
+			authHeader:               "Basic aladdin:opensesame",
+			expectedStatusCode:       401,
+			expectedServiceWasCalled: false,
+			expectedBody:             "No access token provided",
+		},
+		{
+			name:                     "token is too long (should not query the DB)",
+			authHeader:               "Bearer " + strings.Repeat("1", 256),
+			expectedStatusCode:       401,
+			expectedServiceWasCalled: false,
+			expectedBody:             "The access token has expired",
+		},
+	}
```

Comment:
From what I understand, you loop through all values of the authorization header. I do not see that tested here.

Reply:
added the test